src/plugins/es_ui_shared/__packages_do_not_import__/errors/es_error_handler.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ApiError } from '@elastic/elasticsearch';
+import { IKibanaResponse, KibanaResponseFactory } from 'kibana/server';
+
+interface EsErrorHandlerParams {
+  error: ApiError;
+  response: KibanaResponseFactory;
+}
+export const esErrorHandler = ({ error, response }: EsErrorHandlerParams): IKibanaResponse => {
```

Comment:
Nit: can we rename this with a verb? I find that code that uses verb-based method names and noun-based object names is a bit easier to read. For example `handleEsError` would be a little more self-descriptive when reading the consuming code:

```js
} catch (error) {
  return handleEsError({ error, response });
}

Reply:
Great suggestion, @cjcenizal, I agree! Using a verb-based method name here reads almost like a natural language :D